dkc/core/models/user_quota.py

```diff
@@ -1,26 +1,29 @@
 from typing import Type
 
+from django.apps import apps
 from django.contrib.auth.models import User
 from django.db import models
 from django.db.models.signals import post_save
 from django.dispatch import receiver
 from django_girders.models import SelectRelatedManager
 
 
-class Quota(models.Model):
-    allocation = models.BigIntegerField(default=0)
-    user = models.OneToOneField(User, null=True, on_delete=models.CASCADE)
-    # root_set
+class UserQuota(models.Model):
+    user = models.OneToOneField(
+        User, primary_key=True, on_delete=models.CASCADE, related_name='quota'
+    )
+    used = models.PositiveBigIntegerField(default=0)
+    allowed = models.PositiveBigIntegerField()
 
     # The user is needed to stringify, so join it when directly querying for a quota
     objects = SelectRelatedManager('user')
 
     def __str__(self):
-        return f'Quota for <{self.user}>; {self.allocation} bytes'
+        return f'Quota for <{self.user}>; used {self.used} of {self.allowed} bytes'
 
 
 @receiver(post_save, sender=User)
 def create_user_quota(sender: Type[User], instance: User, created: bool, **kwargs):
     if created:
-        quota = Quota(user=instance)
-        quota.save()
+        default_quota = apps.get_app_config('core').DEFAULT_USER_QUOTA
+        UserQuota(user=instance, allowed=default_quota).save()
```

Comment:
Behavior in a signal handler should always be a last resort. We can set the default `allowed` on the field itself, see my comment there.